### ISSUE TITLE
Cannot split line geometry at vertex if the geometry has Z dimension

### ISSUE THREAD

**Issue by komima** [qgis/QGIS#49403]
### What is the bug or the crash?

Trying to split features on a LineStringZ-type layer by using the "split features" tool with snapping on vertices enabled (or using python api `QgsVectorlayer.splitFeatures` and passing a single point that is a vertex) will not split anything (even though python api returns success status)

### Steps to reproduce the issue

Run this snippet in python console to test non-z layer, observe success (feature count == 2):
```python
non_z_layer = QgsMemoryProviderUtils.createMemoryLayer(
    "has-no-z",
    QgsFields(),
    QgsWkbTypes.Type.LineString,
    QgsCoordinateReferenceSystem("EPSG:3067"),
)
assert non_z_layer.startEditing()
assert non_z_layer.addFeature(
    QgsVectorLayerUtils.createFeature(non_z_layer, QgsGeometry.fromWkt("LINESTRING(0 0, 1 1, 2 2)"), {})
)
assert non_z_layer.splitFeatures([QgsPointXY(1, 1)]) == Qgis.GeometryOperationResult.Success
assert non_z_layer.featureCount() == 2
```

Run this snippet in python console to test z-layer, observe failure to split (feature count == 1) but success from api call:
```python
z_layer = QgsMemoryProviderUtils.createMemoryLayer(
    "has-z",
    QgsFields(),
    QgsWkbTypes.Type.LineStringZ,
    QgsCoordinateReferenceSystem("EPSG:3067"),
)
assert z_layer.startEditing()
assert z_layer.addFeature(
    QgsVectorLayerUtils.createFeature(z_layer, QgsGeometry.fromWkt("LINESTRINGZ(0 0 0, 1 1 0, 2 2 0)"), {})
)

# this indicates success
assert z_layer.splitFeatures([QgsPointXY(1, 1)]) == Qgis.GeometryOperationResult.Success

# cannot call this signature due to signature mismatch, see #37821
# z_layer.splitFeatures([QgsPoint(1, 1, 0)])

# this does not work either since plain QgsLineString is seen as an iterable of QgsPoint?
# z_layer.splitFeatures(QgsLineString([QgsPoint(1, 1, 0)]))

# passing explicit defaults matches the correct signature
# this indicates success status, split points is empty list
result, points = z_layer.splitFeatures(QgsLineString([QgsPoint(1, 1, 0)]), False, False)
assert result == Qgis.GeometryOperationResult.Success
assert points == []

# still have one feature only
assert z_layer.featureCount() == 1
```

Also reproducible in GUI by creating a temp layer with linestring + z dimension type, enabling snapping on vertices and clicking a vertex with split features tool: it shows no warning (since api indicates success), but no split is done.

### Versions

QGIS version
3.22.8-Białowieża
QGIS code revision
8d5e9761df
Qt version
5.15.3
Python version
3.9.5
Compiled against GDAL/OGR
3.5.0
Running against GDAL/OGR
3.5.1
PROJ version
9.0.1
EPSG Registry database version
v10.064 (2022-05-19)
GEOS version
3.10.3-CAPI-1.16.1
SQLite version
3.38.1
PDAL version
2.3.0
PostgreSQL client version
14.3
SpatiaLite version
5.0.1
QWT version
6.1.6
QScintilla2 version
2.13.1
OS version
Windows 10 Version 2009

Active Python plugins

### Supported QGIS version

- [X] I'm running a supported QGIS version according to the roadmap.

### New profile

- [X] I tried with a new QGIS profile

### Additional context

QgsPoint/QgsPointXY signature mismatch is similar to issue https://github.com/qgis/QGIS/issues/37821, which forces to pass QgsPointXY's (z values are dropped), but even by passing explicit defaults to match the QgsCurve-signature (and to preserve z values) splitting does not work.

**Comment by uclaros** [comment]
The splitting logic has been revamped in 3.26 to fix some old standing bugs (see #47306) and apparently has also fixed this!
The use case could be added as a regression test though.

**Comment by agiudiceandrea** [comment]
I can confirm that the issue occurs with QGIS 3.22.10 while it doesn't occur with QGIS 3.26.2.

**Comment by agiudiceandrea** [comment]
Closing, as it is fixed in QGIS >= 3.26 and there will be no more QGIS 3.22 versions according to the road-map.
